esup-child.el

```diff
@@ -189,17 +188,23 @@ a complete result.")
     (setq str (replace-match "" t t str)))
   str)
 
+(defun esup-child-s-pad-left (len padding s)
+  "If S is shorter than LEN, pad it with PADDING on the left."
+  (let ((extra (max 0 (- len (length s)))))
+    (concat (make-string extra (string-to-char padding))
+            s)))
+
 (defun esup-child-unindent (str)
   "Remove common leading whitespace from each line of STR.
 If STR contains only whitespace, return an empty string."
-  (let* ((lines (s-lines str))
-         (non-whitespace-lines (seq-filter (lambda (s) (< 0 (length (s-trim-left s))))
+  (let* ((lines (split-string str "\\(\r\n\\|[\n\r]\\)"))
```

Comment:
This line is exactly the same as `s-lines`. You can test it with the snippet below.

```el
(message "%s" (split-string "hell\nworld\n" "\\(\r\n\\|[\n\r]\\)"))
(message "%s" (s-lines "hell\nworld\n"))
```

They have the same result.